File: android/src/main/java/tech/linjiang/flutter/wx/FlutterWxPlugin.java
package tech.linjiang.flutter.wx;

import io.flutter.plugin.common.MethodCall;
import io.flutter.plugin.common.MethodChannel;
import io.flutter.plugin.common.MethodChannel.MethodCallHandler;
import io.flutter.plugin.common.MethodChannel.Result;
import io.flutter.plugin.common.PluginRegistry.Registrar;

/**
 * FlutterWxPlugin
 */
public class FlutterWxPlugin implements MethodCallHandler {
    /**
     * Plugin registration.
     */
    public static void registerWith(Registrar registrar) {
        final MethodChannel channel = new MethodChannel(registrar.messenger(), "tech.linjiang.flutter.wx/plugin");
        channel.setMethodCallHandler(new FlutterWxPlugin());
        Utils.init(registrar.context());
    }

    @Override
    public void onMethodCall(MethodCall call, Result result) {
        WxMiddleware.get().handleEvent(call, result);
    }
}

File: ios/Classes/FlutterWxPlugin.m
#import "FlutterWxPlugin.h"

@implementation FlutterWxPlugin
+ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
  FlutterMethodChannel* channel = [FlutterMethodChannel
      methodChannelWithName:@"tech.linjiang.flutter.wx/plugin"
            binaryMessenger:[registrar messenger]];
  FlutterWxPlugin* instance = [[FlutterWxPlugin alloc] init];
  [registrar addMethodCallDelegate:instance channel:channel];
}

- (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {

}

@end
